File: EmbASP_Desktop_BlocksWorld_CSharp/Stack.cs
using it.unical.mat.embasp.languages;
using System;
using System.Collections.Generic;
using System.Text;

namespace BlocksWorld_SPD
{
    [Id("stack")]
    class Stack
    {
        [Param(0)]
        private string block1;

        [Param(1)]
        private string block2;

        public Stack()
        {
            this.block1 = "";
            this.block2 = "";
        }

        public Stack(string block1, string block2)
        {
            this.block1 = block1;
            this.block2 = block2;
        }

        public string getBlock1()
        {
            return this.block1;
        }

        public string getBlock2()
        {
            return this.block2;
        }

        public void setBlock1(string block1)
        {
            this.block1 = block1;
        }

        public void setBlock2(string block2)
        {
            this.block2 = block2;
        }

        public override string ToString()
        {
            return "stack(block1 = " + block1 + ", block2 = " + block2 + ")";
        }
    }
}

File: EmbASP_Desktop_BlocksWorld_Java/src/it/mat/unical/embasp/Stack.java
package it.mat.unical.embasp;

import it.unical.mat.embasp.languages.Id;
import it.unical.mat.embasp.languages.Param;

@Id("stack")
public class Stack {

	public Stack() {
		block1 = new String();
		block2 = new String();
	}

	@Param(0)
	private String block1;

	@Param(1)
	private String block2;

	public String getBlock1() {
		return block1;
	}

	public void setBlock1(String block1) {
		this.block1 = block1;
	}

	public String getBlock2() {
		return block2;
	}

	public void setBlock2(String block2) {
		this.block2 = block2;
	}

	@Override
	public String toString() {
		return "stack(block1 = " + block1 + ", block2 = " + block2 + ")";
	}

}
